onnxruntime/test/providers/cpu/tensor/pad_test.cc

```diff
@@ -301,5 +305,120 @@ TEST(TensorOpTest, Pad_Constant_2D_double) {
                       "edge");
 }
 
+/*
+Example numpy for testing behavior
+
+import numpy as np
+
+a = np.zeros((2, 0))
+
+b = np.pad(a, 1, 'constant')
+print('constant')
+print(b)
+print(b.shape)
+
+c = np.pad(a, ((1,1),(0,0)), 'reflect')  # allowed if we don't pad the dim with '0'. error otherwise
+print('reflect')
+print(c)
+print(c.shape)
+
+d = np.pad(a, 1, 'edge')
+print('edge')
+print(d)
+print(d.shape)
+
+Output:
+
+constant
+[[0. 0.]
+ [0. 0.]
+ [0. 0.]
+ [0. 0.]]
+(4, 2)
+reflect
+[]
+(4, 0)
+edge
+[]
+(4, 0)
+*/
+
+// test handling of input with a 0 for a dimension
+TEST(TensorOpTest, Pad_Constant_DimWithZeroInput) {
+  RunAllOpsetAllDomainPadTests({0},  // 1D
```

Comment:
(optional): maybe add a test with output shape containing 0 even after padding some dims in constant mode. For ex: input_shape = {0}, pads = {-1, 1}. output_shape = {0} (and hence the output buffer is empty as well)

Reply:
added